### DISCUSSION TITLE
Add hook to ps_emailalerts module

### DISCUSSION THREAD

**Discussion by musaffar-patel** [PrestaShop/PrestaShop#33393] [Ideas]
#### Is your feature request related to a problem?

At the moment I have to rely on modifying the ps_emailalerts module for clients which of course is a bad idea as updates to the module  would see the changes over written.

#### Describe the solution you'd like

It would be great to have a hook in the ps_emailalerts module to allow modules to display custom information in the email.  The hook should receive the order ID as a minimum,

Kind Regards
Musaffar

**Comment by hibatallahAouadni** [comment]
Hello @musaffar-patel 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by Hlavtox** [comment]
Hi @musaffar-patel, you can use `actionEmailSendBefore` for this purpose, you can add any variable you want into the email. Example.

```php
public function hookActionEmailSendBefore($params)
{
    if ($params['template'] == 'new_order') {
        $reference = (string)$params['templateVars']['{order_name}'];
        $order = Order::getByReference($reference)->getFirst();
        // some code
        $params['templateVars']['{new_variable}'] = 'Blablabla';
    }
}
```
